### ISSUE TITLE
App freezes once In app purchase is successful 

### ISSUE THREAD

**Issue by jack-sparroow** [RevenueCat/react-native-purchases#440]
- [✅] I have updated Purchases SDK to the latest version
- [✅] I have read the [Contribution Guidelines](https://github.com/RevenueCat/react-native-purchases/blob/main/CONTRIBUTING.md)
- [✅] I have searched the [Community](https://community.revenuecat.com)
- [✅] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [✅] I have searched for [existing Github issues](https://github.com/RevenueCat/react-native-purchases/issues)

**Describe the bug**

After showing products in my application I enable user to purchase them. Here, as I am working on IOS I am using `Iphone 13` to test IAP with sandbox user credentials. Once user clicks on button, I call the following function 

```js
 // purchase
  const purchase = async (incomingPackage) => {
    try {
      Purchases.setDebugLogsEnabled(true);
      console.log("initiated purchase");
      const { customerInfo, productIdentifier } =
        await Purchases.purchasePackage(incomingPackage);
        // Unlock that great "pro" content
        console.log("customerInfo", customerInfo);
        console.log("productIdentifier", productIdentifier);
      console.log("done puchase");
    } catch (e) {
      console.log(JSON.stringify(e, null, 4));
      console.log("err", e);
    } finally {
      console.log("in finally block");
    }
  };

```
with package_object returned by function `Purchases.getOfferings()`.

Once this function is called, it smoothly opens native popup and once I added the credentials purchase is successful but, Once I get this popup

![image](https://user-images.githubusercontent.com/93463215/192506888-4e23e3c2-fe06-436a-82b3-ef24e8dd122e.png)

App is totally freezed. I cannot do any activities on App whatsoever 

1. Environment
   1. Platform: IOS
   2. Revenue Cat SDK version: "react-native-purchases": "^5.0.2",
   3. OS version: Mac OS Monetery 12.6
   4. Xcode/Android Studio version: Xcode 14
   5. React Native version:  "react-native": "0.69.5",
   7. How widespread is the issue. Percentage of devices affected. not sure

**Comment by RCGitBot** [comment]
👀 [SDKONCALL-134] We've just linked this issue to our internal tracker and notified the team. Thank you for reporting, we're checking this out!

**Comment by jackz7** [comment]
https://community.revenuecat.com/sdks-51/application-get-s-stuck-once-we-get-you-are-all-set-from-apple-app-store-2078

**Comment by codykerns** [comment]
Hey @jackz7 - do you mean that it freezes so that you can't select "OK", or after you select "OK" your app is frozen?

Are you able to reproduce the issue with the [MagicWeather](https://github.com/RevenueCat/react-native-purchases/tree/main/examples/MagicWeather) sample app?

**Comment by jackz7** [comment]
@codykerns it is the second option. After I selected "OK" I can't use my app, even left swipe or right swipe gesture also doesn't work.

For more info:

`package.json` 

```json 
{
  "main": "index.js",
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "web": "expo start --web",
    "start": "expo start --dev-client",
    "prebuild": "expo prebuild"
  },
  "dependencies": {
    "@expo-google-fonts/cinzel": "^0.2.2",
    "@expo-google-fonts/cinzel-decorative": "^0.2.2",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-navigation/stack": "^6.1.1",
    "axios": "^0.25.0",
    "expo": "~46.0.9",
    "expo-app-loading": "~2.1.0",
    "expo-auth-session": "~3.7.1",
    "expo-av": "~12.0.4",
    "expo-file-system": "~14.1.0",
    "expo-gl": "~11.4.0",
    "expo-random": "~12.3.0",
    "expo-splash-screen": "~0.16.2",
    "expo-status-bar": "~1.4.0",
    "expo-three": "^6.0.1",
    "expo-three-orbit-controls": "^2.0.0",
    "expo-updates": "~0.14.5",
    "gsap-rn": "^0.0.15",
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-native": "0.69.5",
    "react-native-elements": "^3.4.2",
    "react-native-gesture-handler": "^2.6.1",
    "react-native-phone-number-input": "^2.1.0",
    "react-native-purchases": "^5.0.2",
    "react-native-reanimated": "~2.9.1",
    "react-native-safe-area-context": "^4.3.3",
    "react-native-screens": "~3.15.0",
    "react-native-vector-icons": "^9.0.0",
    "react-native-web": "~0.18.7",
    "react-native-webview": "11.23.0",
    "react-navigation": "^4.4.4",
    "react-navigation-stack": "^2.10.4",
    "react-navigation-tabs": "^2.11.2",
    "three": "^0.127.0"
  },
  "devDependencies": {
    "@babel/core": "^7.18.6"
  },
  "private": true,
  "name": "Sample App",
  "version": "1.0.0"
}

```

**Comment by jackz7** [comment]
My app uses `Three.js` for animations. Is this could be culprit ?

What do you think ?

**Comment by jackz7** [comment]
Xcode Logs 

```bash

2022-09-28 17:22:20.284753+0530 SampleApp[29488:1983220] [javascript] initiated purchase
2022-09-28 17:22:20.288542+0530 SampleApp[29488:1982824] [Purchases] - DEBUG: ℹ️ Vending Offerings from cache
2022-09-28 17:22:20.288844+0530 SampleApp[29488:1982824] [Purchases] - INFO: 💰 Purchasing Product 'spirit_traveller_monthly_1' from package in Offering 'Plans'
2022-09-28 17:22:20.294984+0530 SampleApp[29488:1982824] [Purchases] - DEBUG: ℹ️ PaymentQueue updatedTransaction: spirit_traveller_monthly_1 0
2022-09-28 17:22:20.295202+0530 SampleApp[29488:1982824] [Purchases] - WARN: 🍎‼️ There is a problem with the SKPaymentTransaction missing a transaction date - this is an issue with the App Store. Unix Epoch will be used instead. 
Transactions in the backend and in webhooks are unaffected and will have the correct timestamps. This is a bug in StoreKit 1. To prevent running into this issue on devices running iOS 15+, watchOS 8+, macOS 12+, and tvOS 15+, you can set `usesStoreKit2IfAvailable` to true when calling `configure`.
2022-09-28 17:22:20.295288+0530 SampleApp[29488:1982824] [Purchases] - WARN: 🍎‼️ There is a problem with the SKPaymentTransaction missing a transaction identifier - this is an issue with the App Store.Transactions in the backend and in webhooks are unaffected and will have the correct identifier. This is a bug in StoreKit 1. To prevent running into this issue on devices running iOS 15+, watchOS 8+, macOS 12+, and tvOS 15+, you can set `usesStoreKit2IfAvailable` to true when calling `configure`.
2022-09-28 17:22:38.819729+0530 SampleApp[29488:1982824] [expo] 🟢 Posting 'appBecomesActive' event to registered modules
2022-09-28 17:22:38.825107+0530 SampleApp[29488:1982824] [Purchases] - DEBUG: ℹ️ applicationDidBecomeActive
2022-09-28 17:22:44.606874+0530 SampleApp[29488:1983226] [tcp] tcp_input [C14.1.1.1:3] flags=[R] seq=2505783401, ack=0, win=0 state=LAST_ACK rcv_nxt=2505783401, snd_una=1357774513
2022-09-28 17:25:04.007634+0530 SampleApp[29488:1983778] [ServicesDaemonManager] interruptionHandler is called. -[FontServicesDaemonManager connection]_block_invoke
(lldb) 


```

**Comment by jackz7** [comment]
I am testing this app on physical device i.e. Iphone 13 [IOS 16] and simulator i.e. Iphone 14 Pro max.

When tested on physical device app freezes and no additional logs other than above, but when tested on simulator I get one additional log 

```bash
[StoreKit] <SKPaymentQueue: 0x600000ef7950>: Payment completed with error: Error Domain=ASDErrorDomain Code=500 "Unhandled exception"
UserInfo={NSUnderlyingError=0x600005ffa1c0 {Error Domain=AMSErrorDomain Code=100 "Authentication Failed" UserInfo={NSMultipleUnderlyingErrorsKey=(
"Error Domain=AMSErrorDomain Code=2 \"An unknown error occurred. Please try again.\" UserInfo={NSLocalizedDescription=An unknown error occurred. Please try again.}",
"Error Domain=com.apple.accounts Code=4 \"No auth plugin to verify credentials for accounts of type com.apple.account.iTunesStore.sandbox\"
UserInfo={NSLocalizedDescription=No auth plugin to verify credentials for accounts of type com.apple.account.iTunesStore.sandbox}"
), NSLocalizedDescription=Authentication Failed, NSLocalizedFailureReason=The authentication failed.}}, storefront-country-code=USA, client-environment-type=Sandbox,
NSLocalizedFailureReason=An unknown error occurred, NSLocalizedDescription=Unhandled exception}
```

**Comment by codykerns** [comment]
Hey @sudarshan-dwata! Thanks for the additional logs here. It doesn't seem to indicate why your app is freezing though, unfortunately.

Were you able to reproduce the issue with the [MagicWeather](https://github.com/RevenueCat/react-native-purchases/tree/main/examples/MagicWeather) sample app? That might help to determine if it's related to our SDK, or if it is related to some kind of configuration issue.

**Comment by jack-sparroow** [comment]
Hi @codykerns ! I have fixed the issue by migrating my specific screen from the current navigation stack to another navigation stack. I haven't found the real culprit but I think whenever an animation, 3D rendering library is used in navigation it will cause issues for screen rendering once we proceed with the `In app purchase`.

This issue is not related to revenue-cat-sdk but yeah it does seems to be an issue with the animation libraries themselves.
